### PR TITLE
Format code using rustfmt

### PR DESCRIPTION
**Pull request by marcusklaas**
Rustfmt has matured a bit more, so I gave formatting clippy another try. Considering clippy's deep nesting levels, I was surprised at how readable most of the output was. Still, there were quite a few issues that needed to be fixed up by hand. This has a proven to be a great project to test rustfmt's limits!

### CODE REVIEW

File: src/shadow.rs [hunk 1]
```diff
@@ -99,45 +114,45 @@ fn check_pat(cx: &LateContext, pat: &Pat, init: &Option<&Expr>, span: Span,
...
+                    let name = field.node.name;
+                    let efield = efields.iter()
+                                        .find(|ref f| f.name.node == name)
+                                        .map(|f| &*f.expr);
+                    check_pat(cx, &field.node.pat, &efield, span, bindings);
                 }
             } else {
                 for field in pfields {
-                    check_pat(cx, &field.node.pat, &None, span, bindings);
+                    check_pat(cx, &field.node.pat, init, span, bindings);
```
Comment (marcusklaas) [inline src/shadow.rs]: Huh? Did rustfmt do this? :confused:

File: src/utils.rs [hunk 2]
```diff
@@ -178,48 +198,68 @@ fn trim_multiline_inner(s: Cow<str>, ignore_first: bool, ch: char) -> Cow<str> {
...
+                                &format!("for further information visit \
+                                          https://github.com/Manishearth/rust-clippy/wiki#{}",
+                                         lint.name_lower()))
     }
 }
 
-pub fn span_help_and_lint<T: LintContext>(cx: &T, lint: &'static Lint, span: Span,
-        msg: &str, help: &str) {
+pub fn span_help_and_lint<T: LintContext>(cx: &T,
+                                          lint: &'static Lint,
```
Comment (Manishearth) [inline src/utils.rs]: Is there a rustfmt setting to avoid these being aggressively split across lines? Same for method calls.

Comment (marcusklaas) [inline src/utils.rs]: There is for function definitions, but not for method calls (yet). Does that style have your preference?

Comment (Manishearth) [inline src/utils.rs]: Personally yes. @llogiq ?

File: src/len_zero.rs [hunk 3]
```diff
@@ -30,21 +30,18 @@ impl LintPass for LenZero {
 impl LateLintPass for LenZero {
     fn check_item(&mut self, cx: &LateContext, item: &Item) {
         match item.node {
-            ItemTrait(_, _, _, ref trait_items) =>
-                check_trait_items(cx, item, trait_items),
-            ItemImpl(_, _, _, None, _, ref impl_items) => // only non-trait
```
Comment (killercup) [inline src/len_zero.rs]: Looks like this (and some other) comments went missing.

Comment (Manishearth) [inline src/len_zero.rs]: That comment is in a *very* strange place anyway.
(Should still exist, but should go down a bit)


Rustfmt needs to explicitly handle comments everywhere. Not ideal; but I don't know how to fix it.

File: src/utils.rs (hunk 2 above)
Comment (llogiq) [inline src/utils.rs]: I also dislike too aggressive splitting. I like it if my functions fit into the available vertical screen space and splitting everything sort of messes that up. I usually only split where necessary to keep within the 120-column limit we try to stay in.

Comment (Manishearth) [inline src/utils.rs]: Yeah, that's pretty much exactly my view.

Comment (birkenfeld) [inline src/utils.rs]: FWIW, I agree.

**Comment by Manishearth** [comment]
Redoing as https://github.com/Manishearth/rust-clippy/pull/540, thanks though!

**Pull request closed without merging**